src/coreclr/jit/lower.cpp

```diff
@@ -5190,10 +5190,34 @@ bool Lowering::LowerUnsignedDivOrMod(GenTreeOp* divMod)
         int    postShift;
         bool   simpleMul = false;
 
+        unsigned bits = type == TYP_INT ? 32 : 64;
+        // if the dividend operand is AND or RSZ with a constant then the number of input bits can be reduced
+        if (dividend->OperIs(GT_AND))
+        {
+            size_t maskCns = static_cast<size_t>(
+                dividend->gtGetOp1()->IsCnsIntOrI()
+                    ? dividend->gtGetOp1()->AsIntCon()->IconValue()
+                    : dividend->gtGetOp2()->IsCnsIntOrI() ? dividend->gtGetOp2()->AsIntCon()->IconValue() : 0);
+            if (maskCns)
+            {
+                unsigned maskBits = 1;
+                while (maskCns >>= 1)
+                    maskBits++;
```

Comment:
Nit: could be written as `BitPosition(genFindHighestBit(maskCns)) + 1`.

Reply:
`BitPosition` only supports 32-bit values and `genFindHighestBit` does a loop anyway, so it wouldn't save much.

Reply:
Yes, I know, it was mostly a style suggestion.